File: app/src/main/java/com/physphile/forbot/ClassHelper.java
package com.physphile.forbot;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.view.MenuItem;
import android.widget.CalendarView;

import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.FragmentManager;

import com.google.firebase.auth.FirebaseAuth;
import com.physphile.forbot.olympiads.OlympsAdapter;
import com.physphile.forbot.profile.ProfileMenuDialog;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.util.Calendar;

import static com.physphile.forbot.Constants.AUTH_ACTIVITY_PATH;
import static com.physphile.forbot.Constants.FRAGMENT_DIALOG_PROFILE_TAG;
import static com.physphile.forbot.Constants.NEWS_CREATE_ACTIVITY_PATH;

public class ClassHelper {
    private Activity activity;
    private OlympsAdapter olympsAdapter;
    private FragmentManager fragmentManager;
    private CalendarView calendarView;
    public Toolbar.OnMenuItemClickListener onMenuItemClickListener = new Toolbar.OnMenuItemClickListener() {
        @Override
        public boolean onMenuItemClick(MenuItem item) {
            switch (item.getItemId()) {
                case R.id.profile:
                    if (FirebaseAuth.getInstance().getCurrentUser() != null) {
                        DialogFragment profileDialog = new ProfileMenuDialog();
                        profileDialog.show(fragmentManager, FRAGMENT_DIALOG_PROFILE_TAG);
                    } else {
                        activity.startActivity(new Intent(AUTH_ACTIVITY_PATH));
                    }
                    break;
                case R.id.createNews:
                    activity.startActivity(new Intent(NEWS_CREATE_ACTIVITY_PATH));
                    break;
                case R.id.set_today:
                    olympsAdapter.clearItems();
                    calendarView.setDate(Calendar.getInstance().getTime().getTime());
                    break;
            }
            return false;
        }
    };

    public ClassHelper(Activity _activity) {
        this.activity = _activity;
    }

    public ClassHelper(Activity _activity, FragmentManager _fragmentManager) {
        this.activity = _activity;
        this.fragmentManager = _fragmentManager;
    }

    public ClassHelper(Activity _activity, FragmentManager _fragmentManager, OlympsAdapter _olympsAdapter, CalendarView _calendarView) {
        this.activity = _activity;
        this.olympsAdapter = _olympsAdapter;
        this.fragmentManager = _fragmentManager;
        this.calendarView = _calendarView;
    }

    public void saveFile(Bitmap bitmap, String name) {
        try {
            FileOutputStream out = activity.openFileOutput(name, Context.MODE_PRIVATE);
            bitmap.compress(Bitmap.CompressFormat.PNG, 100, out);
//            Log.e(LOG_NAME, "файл сохранен");
            out.close();
        } catch (Exception ignored) {
//            Log.e(LOG_NAME, "файл не сохранен");
        }
    }

    public Bitmap readFile(String name) throws FileNotFoundException {
        FileInputStream is = activity.openFileInput(name);
//        Log.e(LOG_NAME, "файл прочитан");
        return BitmapFactory.decodeStream(is);
    }

    public int dpToPx(final float dp) {
        return (int) (dp * activity.getResources().getDisplayMetrics().density);
    }


}

File: app/src/main/java/com/physphile/forbot/ClassHelper.kt
package com.physphile.forbot

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.widget.CalendarView
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.FragmentManager
import com.google.firebase.auth.FirebaseAuth
import com.physphile.forbot.olympiads.OlympsAdapter
import com.physphile.forbot.profile.ProfileMenuDialog
import java.io.FileNotFoundException
import java.util.*

class ClassHelper {
    private var activity: Activity? = null
    private var olympsAdapter: OlympsAdapter? = null
    private var fragmentManager: FragmentManager? = null
    private var calendarView: CalendarView? = null
    @JvmField
    var onMenuItemClickListener = Toolbar.OnMenuItemClickListener { item ->
        when (item.itemId) {
            R.id.profile -> if (FirebaseAuth.getInstance().currentUser != null) {
                val profileDialog: DialogFragment = ProfileMenuDialog()
                profileDialog.show(fragmentManager!!, Constants.FRAGMENT_DIALOG_PROFILE_TAG)
            } else {
                activity!!.startActivity(Intent(Constants.AUTH_ACTIVITY_PATH))
            }
            R.id.createNews -> activity!!.startActivity(Intent(Constants.NEWS_CREATE_ACTIVITY_PATH))
            R.id.set_today -> {
                olympsAdapter!!.clearItems()
                calendarView!!.date = Calendar.getInstance().time.time
            }
        }
        false
    }

    constructor(activity: Activity) {
        this.activity = activity
    }

    constructor(activity: Activity, fragmentManager: FragmentManager?) {
        this.activity = activity
        this.fragmentManager = fragmentManager
    }

    fun saveFile(bitmap: Bitmap, name: String?) {
        try {
            val out = activity!!.openFileOutput(name, Context.MODE_PRIVATE)
            bitmap.compress(Bitmap.CompressFormat.PNG, 100, out)
            //            Log.e(LOG_NAME, "файл сохранен");
            out.close()
        } catch (ignored: Exception) {
//            Log.e(LOG_NAME, "файл не сохранен");
        }
    }

    @Throws(FileNotFoundException::class)
    fun readFile(name: String?): Bitmap {
        val `is` = activity!!.openFileInput(name)
        //        Log.e(LOG_NAME, "файл прочитан");
        return BitmapFactory.decodeStream(`is`)
    }

    fun dpToPx(dp: Float): Int {
        return (dp * activity!!.resources.displayMetrics.density).toInt()
    }
}